presto-main/src/main/java/com/facebook/presto/sql/planner/planPrinter/PlanPrinter.java

```diff
@@ -989,6 +990,7 @@ private void printTableScanInfo(TableScanNode node, int indent)
                     predicate.getDomains().get()
                             .entrySet().stream()
                             .filter(entry -> !outputs.contains(entry.getKey()))
+                            .sorted(Comparator.comparing(o -> o.getKey().toString()))
```

Comment:
This seems like fixing the symptoms. Do you know what is the root cause of non-determinism?

Reply:
I have considered fixing the predicate order at the place they are added as one of the option. But, I did not see a need for the the predicates to be in order anywhere else except in the explain plan. So, I have decided to fix it a that layer.

Are there any requirements/need to have the predicates ordered throughout? Can you please elaborate as to why you think there is a root issue other than explain plan?

Reply:
`TableScanNode#getCurrentConstraint` is used in multiple places in the code. It could be that predicate order matters in those places (e.g: when converting predicate back to filter expression). There are quite sophisticated methods that process `TupleDomain`.

Reply:
@sopel39 Karol, it feels like exploring what other places might need fixing is beyond the scope of this change.

Reply:
Maybe it's just `TupleDomain` itself. For instance I found `TupleDomain#simplify` which uses nondeterministic `HashMap` collector. Another one is: `TupleDomain#intersect` (uses `HashMap`). Another one: `TupleDomain#columnWiseUnion` (uses `HashMap`). Another one: `TupleDomain#transform`. Instead of `HashMap` we could use `LinkedHashMap` that preserves order.

There are probably few such places to be fixed. Then explicit sorting in explain should not be needed and plan itself would be deterministic (not just explain).